software/py/bsg_manycore_link_gen.py

```diff
@@ -117,6 +118,54 @@ def _section(self, name, address, vma_region, lma_region,
 
     return script
 
+
+
+  def get_section_map(self):
+    section_map = [
+      # Format:
+      # <output section>: [<input sections>]
+      ['.text.dram'        , ['.text.interrupt', '.crtbegin','.text','.text.startup','.text.*']],
+      # bsg-tommy: 8 bytes are allocated in.dmem.interrupt for interrupt handler to spill registers.
+      ['.dmem'             , ['.dmem.interrupt', '.dmem','.dmem.*']],
+      ['.data'             , ['.data','.data*']],
+      ['.sdata'            , ['.sdata','.sdata.*','.sdata*','.sdata*.*'
+                              '.gnu.linkonce.s.*']],
+      ['.sbss'             , ['.sbss','.sbss.*','.gnu.linkonce.sb.*','.scommon']],
+      ['.bss'              , ['.bss','.bss*']],
+      ['.tdata'            , ['.tdata','.tdata*']],
+      ['.tbss'             , ['.tbss','.tbss*']],
+      ['.striped.data.dmem', ['.striped.data']],
+      ['.eh_frame.dram'    , ['.eh_frame','.eh_frame*']],
+      ['.rodata.dram'      , ['.rodata','.rodata*','.srodata.cst16','.srodata.cst8',
+                              '.srodata.cst4', '.srodata.cst2','.srodata*']],
+      ['.dram'             , ['.dram','.dram.*']],
+    ]
+
+    if self._move_rodata_to_dmem:
+      # remove .rodata.dram
+      ro_idx = None
+      for i, m in enumerate(section_map):
+        sec = m[0]
+        if sec == ".rodata.dram":
+          ro_idx = i
+          break
+      section_map.pop(ro_idx)
+
+      # add to .dmem
+      for i, m in enumerate(section_map):
+        sec = m[0]
+        if sec == ".dmem":
+          m[1].append(".rodata")
+          m[1].append(".rodata*")
+          m[1].append(".srodata.cst16")
+          m[1].append(".srodata.cst8")
+          m[1].append(".srodata.cst4")
+          m[1].append(".srodata.cst2")
+          m[1].append(".srodata*")
```

Comment:
I can confirm the script is working perfectly in both cases. Maybe simplify the codes with the following?

```
      popped_rodata_dram_map = section_map.pop(ro_idx)

      # add to .dmem
      for i, m in enumerate(section_map):
        sec = m[0]
        if sec == ".dmem":
          for in_sec in popped_rodata_dram_map[1]:
            m[1].append(in_sec)
```

Reply:
okay addressed!